ihmc-perception/src/test/java/us/ihmc/perception/cuda/ExampleCUDAKernel.java

```diff
@@ -0,0 +1,119 @@
+package us.ihmc.perception.cuda;
+
+import org.bytedeco.cuda.cudart.CUstream_st;
+import org.bytedeco.cuda.cudart.dim3;
+import org.bytedeco.cuda.global.cudart;
+import org.bytedeco.javacpp.FloatPointer;
+import us.ihmc.log.LogTools;
+
+import static org.bytedeco.cuda.global.cudart.*;
+
+/**
+ * This is a simple example of a kernel that adds two arrays together. The ways in which kernels can be run vary such that you can create all the JavaCPP Pointers
+ * inside a try-with-resources. The user can create more threads on the GPU. This example attempted to keep things really simple and avoided most of that overhead.
+ */
+public class ExampleCUDAKernel
+{
+
+   private static final String KERNEL_TO_ADD_THE_VALUES_OF_TWO_ARRAYS = """
+         extern "C"
+         
+         __global__
+         void add_arrays(int n, float *x, float *y)
+         {
+            int index = blockIdx.x * blockDim.x + threadIdx.x;
+            int stride = blockDim.x * gridDim.x;
+            for (int i = index; i < n; i += stride)
+               y[i] = x[i] + y[i];
+         }
+         """;
+
+   // Even if you don't create a stream, CUDA will create a default one for you, so we make one ourselves to handle things better
```

Comment:
What's this comment for? It seems like it shouldnt be here

Reply:
True, it should live in the javadoc for whatever method is getting called, I can't remember if we put it there but will take a look

Reply:
Actually looked at this and because we don't use the StreamManager, gonna leave the comment, helpful to the user